compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BTupleType.java

```diff
@@ -175,6 +181,9 @@ public boolean addRestType(BType restType) {
         setCyclicFlag(restType);
         return true;
     }
+    public void setTupleMembers(List<BTupleMember> members) {
+        this.members = members;
+    }
```

Comment:
Why do we need this when we have `setMembers`?

Reply:
`assert members.size() == 0;` will not always be true when we need to set the members

Reply:
I think we have to reset/update member types array here too

Reply:
Adding a method that skips the assertion beats the purpose then, right? Do we know why this assertion was added in the first place? cc @suleka96 

> I think we have to reset/update member types array here too
Yes, and/or in org.wso2.ballerinalang.compiler.semantics.model.types.BTupleType#setMembers also.

Reply:
`setTupleMembers` is only used for https://github.com/ballerina-platform/ballerina-lang/pull/39234/files#diff-75a4933bf7e446d096c3a9ce1c8e8abb41cd5497adba2c1a9c84093240169519R1592 , where member size won't be 0